src/components/MultiselectProblem.tsx

```diff
@@ -1,7 +1,118 @@
+  solutionOptions: string
+}
+
+interface MultiselectFormValues {
+  response: string[]
+}
+
+export const MultiselectProblem = ({
+  solvedCallback, exhaustedCallback, allowedRetryCallback, content, buttonText, solutionOptions,
+  correctResponse, encourageResponse, solution, retryLimit
+}: MultiselectProps): JSX.Element => {
+  const [feedback, setFeedback] = useState('')
+  const [formDisabled, setFormDisabled] = useState(false)
+  const [retriesAllowed, setRetriesAllowed] = useState(0)
+  const solutionArray: string[] = JSON.parse(solution)
+  const parsedOptionValues: string[] = JSON.parse(solutionOptions)
+
+  const schema = Yup.object({
+    response: Yup.array().min(1, 'Please select an answer')
+  })
+
+  const contentRefCallback = useCallback((node: HTMLDivElement | null): void => {
+    if (node != null) {
+      mathifyElement(node)
+    }
+  }, [])
+
+  const clearFeedback = (): void => {
+    setFeedback('')
+  }
+
+  const modifyModel = (values: MultiselectFormValues, e: React.ChangeEvent<HTMLInputElement>): string[] => {
+    const newSet = new Set(values.response)
+    if (e.target.checked) {
+      newSet.add(e.target.value)
+    } else {
+      newSet.delete(e.target.value)
+    }
+    return Array.from(newSet)
+  }
+
+  const generateOptions = (values: MultiselectFormValues, isSubmitting: boolean, setFieldValue: (field: string, value: any, shouldValidate?: boolean) => void): JSX.Element[] => {
+    const options: JSX.Element[] = []
+
+    parsedOptionValues.forEach(val => options.push(
+    <label key={val}>
+      <Field
+      key={val}
+      type="checkbox"
+      name="response"
+      onChange={(e: React.ChangeEvent<HTMLInputElement>) => { clearFeedback(); setFieldValue('response', modifyModel(values, e)) }}
+      disabled={isSubmitting || formDisabled}
+      value={val}>
+      </Field>
+      {val}
+    </label>
+    ))
+
+    return options
+  }
+
+  const compareForm = (form: string[], solution: string[]): boolean => {
+    if (form.length !== solution.length) {
+      return false
+    }
+    for (let i = 0; i < solution.length; i++) {
+      if (!solution.includes(form[i])) {
+        return false
+      }
+    }
+    return true
+  }
+
+  const handleSubmit = async (values: MultiselectFormValues): Promise<void> => {
+    if (compareForm(values.response, solutionArray)) {
+      setFeedback(correctResponse)
+      solvedCallback()
+      setFormDisabled(true)
+    } else if (retryLimit === 0 || retriesAllowed !== retryLimit) {
+      setRetriesAllowed(currRetries => currRetries + 1)
+      setFeedback(encourageResponse)
+      allowedRetryCallback()
+    } else {
+      exhaustedCallback()
+      setFeedback(NO_MORE_ATTEMPTS_MESSAGE)
+      setFormDisabled(true)
+    }
+  }
 
-export const MultiselectProblem = ({ solvedCallback, exhaustedCallback, allowedRetryCallback }: MultiSelectProps): JSX.Element => {
-  return <></>
+  return (
+    <div className="os-raise-bootstrap" ref={contentRefCallback}>
```

Comment:
Again, we don't want to spend too much time on styling in this iteration, but I think here it'd be nice / easy to incorporate use of the `form-check` and accompanying classes documented [here](https://getbootstrap.com/docs/5.0/forms/checks-radios/) so it looks a bit nicer.